### PR TITLE
Added methods for ROCR performance objects

### PR DESCRIPTION
**Pull request by austin3dickey**
Added methods for fortify and autoplot to support performance objects from the ROCR package.

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Thanks Austin! Could you paste some example plots here?

File: tests/testthat/test-performance.R [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  expect_true(is.data.frame(fortified1.2))
+  expect_true(is.data.frame(fortified1.3))
+  expect_true(is.data.frame(fortified1.4))
+  expect_true(is.data.frame(fortified2.1))
+  expect_true(is.data.frame(fortified2.2))
+  expect_true(is.data.frame(fortified2.3))
+  expect_true(is.data.frame(fortified2.4))
+
+  # Check if names are good
+  expect_equal(names(fortified1.1), c('Repetition.Number', 'Cutoff', 'Accuracy'))
```
Comment (terrytangyuan) [inline tests/testthat/test-performance.R]: Let's check for sorted names.

File: tests/testthat/test-performance.R [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+perf2.4 <- ROCR::performance(pred2, 'auc')
+
+
+
+test_that('fortify.performance works for ROCR demo objects', {
+
+  # Fortify our 8 test objects
+  fortified1.1 <- ggplot2::fortify(perf1.1)
+  fortified1.2 <- ggplot2::fortify(perf1.2)
+  fortified1.3 <- ggplot2::fortify(perf1.3)
```
Comment (terrytangyuan) [inline tests/testthat/test-performance.R]: We can probably wrap this into a loop

File: tests/testthat/test-performance.R [hunk 3]
```diff
@@ -0,0 +1,167 @@
+context('test ROCR performance')
+
+# Load demo objects for two scenarios
+library(ROCR)
```
Comment (terrytangyuan) [inline tests/testthat/test-performance.R]: Let's wrap the tests around a skip_if_not_installed

File: R/fortify_performance.R [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+#' @export
+autoplot.performance <- function(object, p = NULL,
+                                 bins = 5, ...) {
+
+  plot.data <- ggplot2::fortify(object)
+  plot.names <- names(plot.data)
+
+  # Three different plots depending on how many columns there are
+
+  if (!(length(plot.names) %in% c(2, 3, 4))) {
```
Comment (terrytangyuan) [inline R/fortify_performance.R]: Perhaps a more informative message like what is malformed exactly?

**Comment by austin3dickey** [comment]
Example plots:
```R
library(ROCR)
data("ROCR.xval")
pred <- ROCR::prediction(ROCR.xval$predictions,
                         ROCR.xval$labels)
for (perfObj in c(performance(pred, 'acc'),
                  performance(pred, 'tpr', 'fpr'),
                  performance(pred, 'auc'))) {
  plot(perfObj, main = 'Base plotter', colorize = TRUE)
  autoplot(perfObj, main = 'autoplot')
}
```
image
image
image
image
Base plot for the third object gets this error: `Error in .plot.performance: Performance object cannot be plotted.`
image

**Comment by terrytangyuan** [comment]
Thanks!! Merged!

**Pull request merged**